Sources/NIOPosix/SocketChannel.swift

```diff
@@ -119,9 +116,25 @@ final class SocketChannel: BaseStreamSocketChannel<Socket> {
                 }
             }
         }
+    }
+
+    override func connectSocket(to address: SocketAddress) throws -> Bool {
+        if try self.socket.connect(to: address) {
+            return true
+        }
+        self.scheduleConnectTimeout()
+        return false
+    }
 
+#if canImport(Darwin) || os(Linux)
```

Comment:
why are these `#if`'d?

Reply:
Because they are `#if`'d in `BaseSocketChannel`.

I'll try and remove them from everywhere and see how I get on.

Reply:
thank you, yeah, let's push the `#if`s as far in as we can. The less `#if`'d code the better